### ISSUE TITLE
ANSI color appears in linux ssh command 

### ISSUE THREAD

**Issue by athoune** [ansible/ansible#5798]
With a Mac, 

```
$ ansible --version
ansible 1.4.4
$ ansible -vvv -i hosts -u root all -m ping

<plop-tests> ESTABLISH CONNECTION FOR USER: root on PORT 22 TO plop-tests
<plop-tests> EXEC /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-1390922238.14-46311863747650 && echo $HOME/.ansible/tmp/ansible-1390922238.14-46311863747650'
<plop-tests> REMOTE_MODULE ping
<plop-tests> PUT /var/folders/nn/qhqqgrrj4q39z5txmy4t6fcc0000gn/T/tmpSkVEZd TO /root/.ansible/tmp/ansible-1390922238.14-46311863747650/ping
<plop-tests> PUT /var/folders/nn/qhqqgrrj4q39z5txmy4t6fcc0000gn/T/tmpHd2s2n TO /root/.ansible/tmp/ansible-1390922238.14-46311863747650/arguments
<plop-tests> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-1390922238.14-46311863747650/ping /root/.ansible/tmp/ansible-1390922238.14-46311863747650/arguments; rm -rf /root/.ansible/tmp/ansible-1390922238.14-46311863747650/ >/dev/null 2>&1'
plop-tests | success >> {
    "changed": false,
    "ping": "pong"
}
```

With a Debian 7

```
$ ansible --version
ansible 1.4.4
$ ansible -vvv -i hosts -u root all -m ping
<plop-tests> ESTABLISH CONNECTION FOR USER: root
<plop-tests> EXEC ['ssh', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/mlecarme/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'Port=22', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', 'plop-tests', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-1390921423.75-253221704312921 && echo $HOME/.ansible/tmp/ansible-1390921423.75-253221704312921'"]
<plop-tests> REMOTE_MODULE ping
<plop-tests> PUT /tmp/tmpoKpX02 TO /root/.ansible/tmp/ansible-1390921423.75-253221704312921/ping
<plop-tests> PUT /tmp/tmpLTrrZO TO /root/.ansible/tmp/ansible-1390921423.75-253221704312921/arguments
<plop-tests> EXEC ['ssh', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/mlecarme/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'Port=22', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', 'plop-tests', "/bin/sh -c '/usr/bin/python \x1b]0;root@plop-tests\x07/root/.ansible/tmp/ansible-1390921423.75-253221704312921/ping \x1b]0;root@plop-tests\x07/root/.ansible/tmp/ansible-1390921423.75-253221704312921/arguments; rm -rf \x1b]0;root@plop-tests\x07/root/.ansible/tmp/ansible-1390921423.75-253221704312921/ >/dev/null 2>&1'"]
plop-tests | FAILED >> {
    "failed": true,
    "msg": "\u001b]0;root@plop-tests\u0007/usr/bin/python: can't open file '\u001b]0': [Errno 2] No such file or directory\r\n/bin/sh: 1: root@plop-tests\u0007/root/.ansible/tmp/ansible-1390921423.75-253221704312921/ping: not found\r\n/bin/sh: 1: root@plop-tests\u0007/root/.ansible/tmp/ansible-1390921423.75-253221704312921/arguments: not found\r\n",
    "parsed": false
}
```

Where does _\x1b]0;_ came from? Is there a trick to remove this dirt from my command line?

**Comment by athoune** [comment]
With a -c paramiko, there is no bug.

Beware of print for debugging. The command is rotten, but if you `print cmd`, you see nothing wrong, but `print [cmd]`show you horrible ANSI escape.

**Comment by dagwieers** [comment]
The reason for this is that ssh is run with pseudo-terminals (option -tt) which it does because sudo by default requires a tty. So everything run within Ansible using pseudo-tty's assumes the output is a terminal, and as a result might wait for input, or produce output intended for terminals. This has been discussed many times before...

**Comment by athoune** [comment]
I love ANSI color, it's not the point. The ticket is for the debian crash. My diagnostic can be wrong, but the error is still here.

I didn't know the tt option, thanks for that.

**Comment by mpdehaan** [comment]
Not sure why the "but print [cmd]" stuff was added, but aren't getting any other reports of this.  I think you may possibly have some color output in your bashrc or something?

**Comment by jimi-c** [comment]
Hi @athoune, going through older issues to see if they've been corrected by the v2 (ansible 2.0) work that is live in devel now. Could you please let us know if you're still seeing this issue? I don't think we've had any other Debian 7 users report this.

**Comment by jimi-c** [comment]
As there has been no follow up on this, we're going to go ahead and close it now.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
